pkg/scheduler/framework/runtime/batch.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	fwk "k8s.io/kube-scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/metrics"
+)
+
+type PodSignatureFunc func(p *v1.Pod) string
+
+// Only needed until we connect signatures
+func noBatchSignatures(p *v1.Pod) string { return "" }
+
+// OpportunisticBatching caches results from filtering and scoring when possible to optimize
+// scheduling of common pods.
+type OpportunisticBatch struct {
+	state          *batchState
+	currPod        batchPodInfo
+	signatureFunc  PodSignatureFunc
+	schedFramework framework.Framework
+}
+
+type batchState struct {
+	signature    string
+	sortedNodes  framework.SortedScoredNodes
+	creationTime time.Time
+}
+
+type batchPodInfo struct {
```

Comment:
If we stored here also the cycle number, maybe we could rename it to something indicating it's scheduling cycle data like `cycleContext` or `schedulingCycle` where pod and signature would be a part of such a context.

```suggestion
type schedulingCycle struct {
```